include/inexor/vulkan-renderer/tools/cla_parser.hpp

```diff
@@ -1,123 +1,100 @@
 ﻿#pragma once
 
-#include <spdlog/spdlog.h>
-
+#include <cstdint>
 #include <optional>
 #include <string>
 #include <unordered_map>
+#include <utility>
 #include <vector>
 
 namespace inexor::vulkan_renderer::tools {
 
-/// @brief Defines the type of an accepted command line argument.
-enum class CommandLineArgumentType {
-    NONE,
-    STRING,
-    INT64,
-    UINT32,
-    BOOL,
-    // TODO: Add more
-};
+/// @brief A command line argument template class
+class CommandLineArgumentTemplate {
+    const std::string name;
```

Comment:
Better use `argument` instead of `name`. (Same goes for the parameters in the methods).